src/Kucipong/Handler/Admin.hs

```diff
@@ -4,90 +4,119 @@ module Kucipong.Handler.Admin where
+adminUrlPrefix :: Path '[] 'Open
 adminUrlPrefix = "admin"
 
-loginPageR :: Path '[]
-loginPageR = "login"
+loginR :: Path '[] 'Open
+loginR = "login"
 
-doLoginR :: Path '[LoginToken]
-doLoginR = "login" <//> var
+doLoginR :: Path '[LoginToken] 'Open
+doLoginR = loginR <//> var
 
-loginPage
+storeCreateR :: Path '[] 'Open
+storeCreateR = "store" <//> "create"
+
+loginGet
     :: forall ctx m
      . ( MonadIO m
        )
     => ActionCtxT ctx m ()
-loginPage =
+loginGet =
     $(renderTemplateFromEnv "adminUser_login.html") $ fromPairs []
 
 -- | Login an admin.  Take the admin's 'LoginToken', and send them a session
 -- cookie.
-doLogin
+doLoginGet
     :: forall ctx m
      . ( MonadIO m
        , MonadKucipongCookie m
        , MonadKucipongDb m
        , MonadTime m
        )
     => LoginToken -> ActionCtxT ctx m ()
-doLogin loginToken = do
+doLoginGet loginToken = do
     maybeAdminLoginTokenEntity <- dbFindAdminLoginToken loginToken
     (Entity (AdminLoginTokenKey (AdminKey adminEmail)) adminLoginToken) <-
         fromMaybeM noAdminLoginTokenError maybeAdminLoginTokenEntity
     -- check date on admin login token
     now <- currentTime
     let (LoginTokenExpirationTime expirationTime) =
-            adminLoginToken ^. adminLoginTokenExpirationTime
+            adminLoginTokenExpirationTime adminLoginToken
     when (now > expirationTime) tokenExpiredError
     setAdminCookie adminEmail
     redirect $ renderRoute root
   where
     noAdminLoginTokenError :: ActionCtxT ctx m a
     noAdminLoginTokenError =
-        redirect . renderRoute $ adminUrlPrefix <//> loginPageR
+        redirect . renderRoute $ adminUrlPrefix <//> loginR
 
     tokenExpiredError :: ActionCtxT ctx m a
     tokenExpiredError =
-        redirect . renderRoute $ adminUrlPrefix <//> loginPageR
+        redirect . renderRoute $ adminUrlPrefix <//> loginR
 
 -- | Return the store create page for an admin.
-storeCreate
+storeCreateGet
     :: forall xs n m
      . ( ContainsAdminSession n xs
        , MonadIO m
        , MonadLogger m
        )
     => ActionCtxT (HVect xs) m ()
-storeCreate = do
+storeCreateGet = do
     (AdminSession email) <- getAdminEmail
     $(renderTemplateFromEnv "adminUser_admin_store_create.html") $ fromPairs
         [ "adminEmail" .= email ]
 
+storeCreatePost
+    :: forall xs n m
+     . ( ContainsAdminSession n xs
+       , MonadIO m
+       , MonadKucipongDb m
+       , MonadKucipongSendEmail m
+       , MonadLogger m
+       )
+    => ActionCtxT (HVect xs) m ()
+storeCreatePost = do
+    (AdminSession email) <- getAdminEmail
+    (AdminStoreCreateForm storeEmailParam) <- getReqParam
+    (Entity storeEmailKey storeEmail) <- dbCreateStoreEmail storeEmailParam
+    (Entity storeLoginTokenKey storeLoginToken) <-
+        dbCreateStoreMagicLoginToken storeEmailKey
+    sendStoreLoginEmail
+        (storeEmailEmail storeEmail)
+        (storeLoginTokenLoginToken storeLoginToken)
+    -- TODO: Where to redirect this?
```

Comment:
Redirect to the store create page, please.
I'll implement to show user as a pop up message on the page that the user has successfully created the store user, later.